src/lib/address/address.ts

```diff
@@ -151,7 +183,17 @@ export type ParseAddressInputReturn = {
 }
 export type ParseAddressInput = (args: ParseAddressInputArgs) => Promise<ParseAddressInputReturn>
 
-export const parseAddressInput: ParseAddressInput = async args => {
+export const parseAddressInput: ParseAddressInput = async ({ value, chainId }) => {
+  const decodedUrlDatas = parseUrlByChainId(chainId, value)
+
+  const args: ParseAddressInputArgs = {
```

Comment:
as discussed with @gomesalexandre this is somewhat *shadowy* in that we're kinda but not really mutating input args here

it would be nicer if we could move this ternary logic up into the `parseUrlByChainId` function and then do something like

`const parsedArgs = parseUrlByChainId(args)`

and then all the functions below can consume it